File: ARCHIVED_FILES/CONVERTERS/convert_webp_to_jpg.py
#!/usr/bin/env python3
import os
import sys
from pathlib import Path

try:
    from PIL import Image
except ImportError:
    print("Installing Pillow...")
    os.system("pip install pillow -q")
    from PIL import Image

folder = sys.argv[1] if len(sys.argv) > 1 else "toto"
directory = f"public/images/products/{folder}"

if not os.path.isdir(directory):
    print(f"Error: Folder not found - {directory}")
    sys.exit(1)

print(f"Converting WebP to JPG in: {directory}")

webp_files = list(Path(directory).glob("*.webp"))

if not webp_files:
    print(f"No WebP files found in {directory}")
    sys.exit(0)

converted = 0
failed = 0

for webp_file in webp_files:
    try:
        jpg_file = webp_file.with_suffix('.jpg')
        filename = webp_file.name
        
        # Open WebP and convert to JPG
        image = Image.open(str(webp_file)).convert('RGB')
        image.save(str(jpg_file), 'JPEG', quality=85)
        
        # Delete WebP file
        webp_file.unlink()
        
        print(f"✓ Converted: {filename}")
        converted += 1
        
    except Exception as e:
        print(f"✗ Failed to convert: {webp_file.name} - {str(e)}")
        failed += 1

print(f"\nConversion completed!")
print(f"Converted: {converted} files")
if failed > 0:
    print(f"Failed: {failed} files")

sys.exit(0)

File: app/Console/Commands/ConvertWebpToJpg.php
<?php

namespace App\Console\Commands;

use Illuminate\Console\Command;

class ConvertWebpToJpg extends Command
{
    protected $signature = 'images:convert-webp-to-jpg {--folder=toto}';
    protected $description = 'Convert WebP images to JPG format';

    public function handle()
    {
        $folder = $this->option('folder');
        $directory = public_path("images/products/{$folder}");

        if (!is_dir($directory)) {
            $this->error("Folder not found: $directory");
            return 1;
        }

        $this->info("Converting WebP to JPG in: $directory");

        $webpFiles = glob($directory . '/*.webp');
        
        if (empty($webpFiles)) {
            $this->info("No WebP files found in $directory");
            return 0;
        }

        $converted = 0;
        $failed = 0;

        foreach ($webpFiles as $webpFile) {
            try {
                $jpgFile = str_replace('.webp', '.jpg', $webpFile);
                $filename = basename($webpFile);
                
                // Convert using GD Library
                $image = imagecreatefromwebp($webpFile);
                if (!$image) {
                    throw new \Exception("Failed to load WebP file");
                }

                // Save as JPG with 80% quality
                imagejpeg($image, $jpgFile, 80);
                imagedestroy($image);

                // Update database
                $oldPath = "images/products/{$folder}/" . basename($webpFile);
                $newPath = "images/products/{$folder}/" . basename($jpgFile);
                
                \DB::table('product_images')
                    ->where('image_path', $oldPath)
                    ->update(['image_path' => $newPath]);

                // Delete WebP file
                unlink($webpFile);

                $this->line("✓ Converted: $filename");
                $converted++;

            } catch (\Exception $e) {
                $this->error("✗ Failed to convert: " . basename($webpFile) . " - " . $e->getMessage());
                $failed++;
            }
        }

        $this->info("\nConversion completed!");
        $this->info("Converted: $converted files");
        if ($failed > 0) {
            $this->warn("Failed: $failed files");
        }

        return 0;
    }
}
?>
